net/xia/ppal_ether/main.c

```diff
@@ -298,8 +329,11 @@ static int main_newroute(struct xip_ppal_ctx *ctx, struct fib_xid_table *xtbl,
 	memmove(mether->ha, cfg->xfc_lladdr, cfg->xfc_lladdr_len);
 	attach_neigh_to_interface(mether);
 
-	WARN_ON(ether_rt_iops->fxid_add_locked(&bucket, xtbl,
-					       &mether->xem_common));
+	xia_ether_header_cache(mether);
```

Comment:
Style: I'd call `xia_ether_header_cache()` just before `attach_neigh_to_interface()` because `xia_ether_header_cache()` only edits the fields of `mether`, whereas `attach_neigh_to_interface()` adds `mether` to the global list of main Ethernet xia fibs associated to a given interface.